src/Analyzers/CSharp/Tests/Copilot/CSharpImplementNotImplementedExceptionDiagnosticAnalyzerTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Copilot.UnitTests;
+
+using VerifyCS = CSharpCodeFixVerifier<
+    CSharpImplementNotImplementedExceptionDiagnosticAnalyzer,
+    EmptyCodeFixProvider>;
+
+public class CSharpImplementNotImplementedExceptionDiagnosticAnalyzerTests
+{
+    [Fact]
+    public async Task TestThrowNotImplementedExceptionInStatement()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+            using System;
+
+            class C
+            {
+                void M()
+                {
+                    {|IDE3000:throw new NotImplementedException();|}
+                }
+            }
+            """,
+            LanguageVersion = LanguageVersion.CSharp11,
+            ReferenceAssemblies = ReferenceAssemblies.Net.Net60,
+        }.RunAsync();
+    }
+
+    [Fact]
+    public async Task TestThrowNotImplementedExceptionInExpression()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+            using System;
+
+            class C
+            {
+                int P => {|IDE3000:throw new NotImplementedException()|};
```

Comment:
add test where it's in a field, constructor, destructor, indexer, event and operator.